File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int gcd(int a, int b) {
    int tmp;
    while (a % b) {
        tmp = a % b;
        a = b;
        b = tmp;
    }
    return b;
}

int solution(vector<int> arr) {
    int answer = arr[0];
    for (int i = 1; i < arr.size(); i++) {
        int g = gcd(answer, arr[i]);
        answer = answer * arr[i] / g;
    }
    return answer;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;

class Solution {
    
    private int gcd(int n1, int n2) {
        int a = n1, b = n2;
        while (a % b != 0) {
            int tmp = a % b;
            a = b;
            b = tmp;
        }
        return n1 * n2 / b;
    }
    
    public int solution(int[] arr) {
        Integer[] newArr = Arrays.stream(arr).boxed().toArray(Integer[]::new);
        Arrays.sort(newArr, Collections.reverseOrder());
        int res = newArr[0];
        for (int i = 1; i < newArr.length; i++)
            res = gcd(res, newArr[i]);
        return res;
    }
}